can.py

```diff
@@ -42,17 +42,17 @@ def relog( self, prefix ):
   def read( self, numChars ):
     s = self._com.read( numChars )
     for ch in s:
-      self._logFile.write( chr(0x01) )
-      self._logFile.write( ch )
+      self._logFile.write( bytes([0x01]) )
+      self._logFile.write( bytes([ch]) )
     self._logFile.flush()
     return s
 
   def write( self, chars ):
     for ch in chars:
-      self._logFile.write( chr(0x00) )
-      self._logFile.write( ch )
+      self._logFile.write( bytes([0x00]) )
+      self._logFile.write( bytes([ord(ch)]) )
     self._logFile.flush()
-    self._com.write( chars )
+    self._com.write( bytes([ord(x) for x in chars]) )
```

Comment:
I don't understand this line. What is it trying to do? Is `chars` unicode string? If yes, what about `chars.encode('ascii')`? But `chars` should not be unicode string - it should be bytes already, right? Or is this the part of code that should be ignored?

Reply:
Thanks - this was my first Python3 code, where I globally changed strings to byte arrays (without clear understanding). The callers part are really strings and it should be rewritten into bytearray - I believe that at that time I wanted minimal change. Thanks for pointing this out - I will probably rewrite this can.py completely.

Reply:
p.s. the input "strings" are binary, but coded like strings, i.e. `chars.encode('ascii')` would not work here:
```
>>> chars = '\xff'
>>> ord(chars[0])
255
>>> chars.encode('ascii')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeEncodeError: 'ascii' codec can't encode character '\xff' in position 0: o
rdinal not in range(128)
```

Reply:
So... the conclusion is... can.py (and others) should work with `bytes`/`bytearray`, right? So no more `ord` and friends, no more binary data inside `str` etc?